server/services/offer.service.js

```diff
@@ -5,11 +5,29 @@ import httpStatus from '../helpers/httpStatus';
 import { OFFER_STATUS } from '../helpers/constants';
 import { getUserById } from './user.service';
 import { getEnquiryById, approveEnquiry } from './enquiry.service';
+import { getPropertyById } from './property.service';
+import { TODAY_DDMMYY } from '../helpers/dates';
 
 const { ObjectId } = mongoose.Types.ObjectId;
 
 export const getOfferById = async (id) => Offer.findById(id).select();
 
+export const generateReferenceCode = async (propertyId) => {
+  const property = await getPropertyById(propertyId);
+  const initials = property.name
+    .match(/\b(\w)/g)
+    .join('')
+    .toUpperCase();
+  const numberSold = await Offer.countDocuments({ propertyId })
+    .then((count) => count + 1)
+    .catch((error) => {
+      throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
+    });
+  const type = `OL${property.houseType.charAt(0)}`;
+  const referenceCode = `${initials}/${type}/0${numberSold}${TODAY_DDMMYY}`;
```

Comment:
This is invalid. This will cause a bug when the number sold is greater or equal to 10 😞 . You need to always think for edge cases before writing your code.

You need to write a function to check if the number is less than 10, append 0 else just output the number. This should be done before you concatenate everything together here.

```suggestion
  const referenceCode = `${initials}/${type}/${numberSold}${TODAY_DDMMYY}`;
```